interfaces/cython/SConscript

```diff
@@ -135,64 +27,113 @@ for f in ['inputs/h2o2.inp', 'inputs/gri30.inp', 'transport/gri30_tran.dat']:
-                 ' $python2_cmd_esc setup2.py build --build-lib=../../build/python2')
-    mod = build(py2env.Command('#build/python2/cantera/__init__.py',
-                               'setup2.py',
-                               build_cmd))
-    env['python2_module'] = mod
-    env['python2_extension'] = ext
-
-    # Use 3to2 to convert examples from Python 3 syntax
-    if env['python_convert_examples']:
-        a = build(py2env.Command('#build/python2/cantera/examples', 'cantera/examples',
-                                 Copy("$TARGET", "$SOURCE")))
-        # In these next few things, the double quotes are necessary
-        # so the command line is processed properly
-        threetotwo_options = ['--no-diff', '-n', '-w', '-x', 'str', '-x', 'print', '-x', 'open',
-                              '-f', 'all', '-f', 'printfunction']
-        threetotwo_options = ', '.join(["'{}'"]*len(threetotwo_options)).format(*threetotwo_options)
-        ex_loc = "'build/python2/cantera/examples'"
-        convert_script = textwrap.dedent("""\
-            $python2_cmd_esc -c "from lib3to2.main import main; main('lib3to2.fixes', [{opts}, {loc}])"
-        """.format(opts=threetotwo_options, loc=ex_loc))
-        b = build(py2env.AddPostAction(a, convert_script))
-        py2env.Depends(mod, b)
-
-    add_dependencies(mod, ext)
-    install_module(py2env['python2_prefix'], py2_version)
+# Get information needed to build the Python module
+script = '\n'.join(("from distutils.sysconfig import *",
+                    "import numpy",
+                    "print(get_config_var('EXT_SUFFIX') or get_config_var('SO'))",
+                    "print(get_config_var('INCLUDEPY'))",
+                    "print(get_config_var('LDLIBRARY'))",
+                    "print(get_config_var('prefix'))",
+                    "print(get_python_version())",
+                    "print(numpy.get_include())"))
+info = getCommandOutput(localenv['python_cmd'], '-c', script)
+module_ext, inc, pylib, prefix, py_version, numpy_include = info.splitlines()[-6:]
+localenv.Prepend(CPPPATH=[Dir('#include'), inc, numpy_include])
+localenv.Prepend(LIBS=localenv['cantera_libs'])
+if localenv['OS'] == 'Darwin':
+    localenv.Append(LINKFLAGS='-undefined dynamic_lookup')
+elif localenv['OS'] == 'Windows':
+    localenv.Append(LIBPATH=prefix+'/libs')
+    if localenv['toolchain'] == 'mingw':
+        localenv.Append(LIBS='python%s' % py_version.replace('.',''))
+        if localenv['OS_BITS'] == 64:
+            localenv.Append(CPPDEFINES='MS_WIN64')
+        # fix for http://bugs.python.org/issue11566
+        localenv.Append(CPPDEFINES={'_hypot':'hypot'})
+elif localenv['OS'] == 'Cygwin':
+    # extract 'pythonX.Y' from 'libpythonX.Y.dll.a'
+    localenv.Append(LIBS=pylib[3:-6])
+
+# Build the Python module
+obj = localenv.SharedObject('#build/temp-py/_cantera', 'cantera/_cantera.cpp')
+ext = localenv.LoadableModule('#build/python/cantera/_cantera%s' % module_ext,
```

Comment:
Since we're changing all of this anyways, can we use the `format` method here (and everywhere else), instead of `%` formatting?

Reply:
In places where there are already other changes, sure. Elsewhere I'm inclined to just leave things as-is, especially since we'll just have the urge to replace things with the `f''`-style formatting once we require Python 3.6 or newer.